salt/state.py

```diff
@@ -838,6 +838,33 @@ def _mod_init(self, low):
                     return
                 self.mod_init.add(low["state"])
 
+    def _aggregate_requisites(self, low, chunks):
+        """
+        Aggregate the requisites
+        """
+        requisites = {}
+        low_state = low["state"]
+        for chunk in chunks:
+            # if the state function in the chunk matches
+            # the state function in the low we're looking at
+            # and __agg__ is True, add the requisites from the
+            # chunk to those in the low.
+            if chunk["state"] == low["state"] and chunk.get("__agg__"):
+                for req in STATE_REQUISITE_KEYWORDS:
+                    if req in chunk:
+                        if req in requisites:
+                            requisites[req].extend(chunk[req])
+                        else:
+                            requisites[req] = chunk[req]
+                for req in STATE_REQUISITE_IN_KEYWORDS:
```

Comment:
**question**: is it requisite (har har) that we loop over these independently, or would `for req in STATE_REQUISITE_KEYWORDS + STATE_REQUISITE_IN_KEYWORDS` (or `itertools.chain(...)`) be clearer here?